src/object/obj_tx.c

```diff
@@ -1564,6 +1772,50 @@ dc_tx_attach(daos_handle_t th, void *args, enum obj_rpc_opc opc,
 		}
 		break;
 	}
+	case DAOS_OBJ_RPC_FETCH: {
+		daos_obj_fetch_t	*fe = dc_task_get_args(task);
+
+		rc = dc_tx_add_read(tx, fe->oh, fe->flags, fe->dkey, fe->nr,
+			fe->nr != 1 ? fe->iods : (void *)&fe->iods[0].iod_name);
+		break;
+	}
+	case DAOS_OBJ_RPC_QUERY_KEY: {
+		daos_obj_query_key_t	*qu = dc_task_get_args(task);
+		daos_key_t		*dkey;
+		uint32_t		 nr;
+
+		if (qu->flags & DAOS_GET_RECX) {
```

Comment:
These flags are not mutually exclusive.   In fact, the most common case is DAOS_GET_RECX | DAOS_GET_DKEY | DAOS_GET_MAX which means it will query the largest DKEY.  So these should be the other way around, I think.   Check if we are querying the dkey first.

Reply:
right, the only thing that's mutually exclusive in the query part is the MIN and MAX.
otherwise you can query for all the types (dkey, akey, recx) in the same call.

Reply:
OK, I will fix it.